src/pudl/io_managers.py

```diff
@@ -83,6 +87,72 @@ def __getitem__(self, idx):
         return self.fk_errors[idx]
 
 
+def get_table_name_from_context(context: OutputContext) -> str:
+    """Retrieves the table name from the context object."""
+    # TODO(rousik): Figure out which kind of identifier is used when.
+    if context.has_asset_key:
+        return context.asset_key.to_python_identifier()
+    return context.get_identifier()
+
+
+class PudlMixedFormatIOManager(ConfigurableIOManager):
+    """Format switching IOManager that supports sqlite and parquet.
+
+    We currently support both sqlite and parquet formats. This io manager
+    allows us to choose which formats should be used for reading and which
+    should be used for reading.
+
+    This should allow us to develop support for parquet format without
+    affecting the existing sqlite-based functionality.
+
+    Switching between reads from sqlite and parquet should allow us
+    to test the two formats for eqivalence during development.
+    """
+
+    # TODO(rousik): the folowing EnvVars should really be booleans but
+    # dagster EnvVar doesn't recognize anything but str/int types here
+    # and AFAIK there's no way to set default value if the env variable
+    # is not set.
+
+    write_to_parquet: bool = bool(EnvVar("PUDL_WRITE_TO_PARQUET").get_value(False))
+    """If true, data will be written to parquet files."""
+
+    read_from_parquet: bool = bool(EnvVar("PUDL_READ_FROM_PARQUET").get_value(False))
+    """If true, data will be read from parquet files instead of sqlite."""
```

Comment:
In short, practical concerns.

I'm treating this as a "feature flag", where we start by having the new functionality (writing to parquet files) off by default, but toggle-able on demand. While toggling this in the dagster UI is useful for local development, in many cases where we might want to test this (e.g. in the CI, nightly builds and other dockerized/remote scenarios), it's very easy to set env variable and pass it to the ETL, and it's much more difficult to be tweaking dagster configuration from the outside IMO. 

This could be hooked into command-line flag, but that is also somewhat clunky and would require ad-hoc wiring to be piped through while env variables are easy and widely supported (both by github actions as well as when running docker image on vm/batch), so it should be fairly easy to toggle this new functionality just where we need it.

Reply:
If all goes well and this feature will end up being enabled everywhere and always, we can drop this toggle and hardcode the new behavior (always on). I'm unsure what would be the timeline for this, given the early stage this is in.

Reply:
If reading from Parquet ends up significantly speeding up the ETL it's easy to imagine this getting turned on immediately.

Once it's there, I'd experiment with using it to speed up the integration tests and data validations (which would mean relying on SQLite primarily for bulk distribution, validation of referential integrity, and maybe other schema checks / constraints that are only implemented by the `to_sqlite()` schema). Since we want to remove `PudlTabl` (probably in the Spring) and that will mean reorganizing the tests, that would probably be a good time to play around with it. I don't know exactly what fraction of the time in the integration tests and data validations is spent reading data, but I suspect it's significant given the way we've wired up `PudlTabl` to pass the read requests through to the database and not cache anything in memory, and I think reading

I can't remember where we were discussing it but I think @bendnorman suggested that since the PUDL ETL script is now a relatively thin wrapper around some slightly nonstandard Dagster structures, we might want to switch to using the Dagster CLI directly, which I think would make these job configurations more straightforward with the Dagster configs.